utils/replicaset/canary.go

```diff
@@ -160,23 +160,23 @@ func CalculateReplicaCountsForCanary(rollout *v1alpha1.Rollout, newRS *appsv1.Re
 		}
 	}
 
-	minAvailableReplicaCount := rolloutSpecReplica - MaxUnavailable(rollout)
+	if GetReplicaCountForReplicaSets(oldRSs) > 0 {
```

Comment:
Note this was changed from `GetAvailableReplicaCountForReplicaSets` to `GetReplicaCountForReplicaSets` to avoid the same problem with older RSs.

Reply:
Also to simplify logic, i move the check to return early if there are older RSs up here, so all the logic below only has to deal with new and stable RS, and not have to factor in number of older RSs.